### ISSUE TITLE
fix broken build as side effect of `ocdev component create --dir`

### ISSUE THREAD

**Issue by containscafeine** [redhat-developer/odo#57]
`ocdev component create nodejs --dir=...` or any other component with `--dir` will return a build that fails and another one that passes. This is how the code has been implemented because we first need to run `oc new-app nodejs~temp-dir` and that directory has no code. This means that the first build will fail. And then we run the second build using `oc start-build --from-dir`.
This is done because there is no way in `oc new-app` cannot take local source code as input but instead detected the `origin` remote from the given repo as input and then pulls it and deploys it.

The output looks like -
```console
$ oc get pods
NAME             READY     STATUS      RESTARTS   AGE
nodejs-1-build   0/1       Error       0          1m
nodejs-1-vst86   1/1       Running     0          1m
nodejs-2-build   0/1       Completed   0          1m
```

This is not a good user experience.
We need to either remove the extra build in ocdev, or we need  a better way of implementing this.

**Comment by containscafeine** [comment]
Take into account the discussion happened at https://github.com/openshift/origin/issues/18571

**Comment by cdrage** [comment]
Is this still an issue @containscafeine ?

**Comment by kadel** [comment]
> I believe this is being fixed in here: #408 right @kadel ?

I believe this was already fixed in https://github.com/redhat-developer/odo/pull/363
